vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/LinkManager.java

```diff
@@ -302,7 +304,79 @@ public IModuleLinkDisplay createModuleLink(String title, ISpace source, float po
         moduleLinkDisplayRepo.save((ModuleLinkDisplay) display);
         return display;
     }
+    
+    @Override
+    public IModuleLinkDisplay editModuleLink(String title, ISpace source, float positionX, float positionY,
+            int rotation, String linkedModuleId, String moduleLinkLabel, String linkId, String moduleLinkDisplayId)
+            throws SpaceDoesNotExistException {
+        
+        source = spaceManager.getSpace(source.getId());
+        if (source == null) {
+            throw new SpaceDoesNotExistException();
+        }
+        Optional<ModuleLink> linkOptional = moduleLinkRepo.findById(linkId);
+        IModuleLink link = linkOptional.get();
+        link.setName(title);
+        IModule module = new Module();
+        module.setId(linkedModuleId);
+        link.setModule(module);
+        moduleLinkRepo.save((ModuleLink) link);
+        Optional<ModuleLinkDisplay> moduleLinkOptional = moduleLinkDisplayRepo.findById(moduleLinkDisplayId);
+        IModuleLinkDisplay display = moduleLinkOptional.get();
```

Comment:
same as above and also, I would first get all objects from the db and set all the values and then save them all. Otherwise, it's a) harder to follow what is changed and saved (it's a lot easier to just find the block of code that saves objects to see which objects are being saved) and also if for example the display object doesn't exist the link object should be changed and there is no db transaction that needs to be rolled back.